runtime/etc/Makefile.mli-shared

```diff
@@ -0,0 +1,140 @@
+# Copyright 2004-2019 Cray Inc.
+# Other additional copyright holders may be indicated within.
+#
+# The entirety of this work is licensed under the Apache License,
+# Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.
+#
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#
+# Makefile.mli-static
+#
+
+include $(CHPL_MAKE_HOME)/runtime/etc/Makefile.include
+
+CHPL_MLI_INCS = -I$(CHPL_ZMQ_LOC)/include
+CHPL_LN_LIB_DIR = $(CHPL_MAKE_RUNTIME_LIB)/$(CHPL_MAKE_LAUNCHER_SUBDIR)
+UNAME = $(shell uname)
+
+.PHONY: checkZMQ buildServer buildClient runTagsCommand
+
+all: checkZMQ buildServer buildClient runTagsCommand FORCE
+
+checkZMQ:
+ifndef CHPL_ZMQ_LOC
+  $(error CHPL_ZMQ_LOC must be set to the location of the ZeroMQ library)
+endif
+
+buildServer: $(TMPSERVERNAME)
+
+buildClient: $(TMPBINNAME)
+
+runTagsCommand:
+	$(TAGS_COMMAND)
+
+$(TMPBINNAME): $(CHPL_CL_OBJS) FORCE
+#
+# Invoke the launcher script, but don't actually link up the launcher binary.
+# Do this by setting CHPL_MAKE_IS_MLI=1.
+#
+	$(MAKE) -f $(CHPL_MAKE_HOME)/runtime/etc/Makefile.launcher all \
+		CHPL_MAKE_HOME=$(CHPL_MAKE_HOME) \
+		TMPBINNAME=$(TMPSERVERNAME) \
+		BINNAME=$(SERVERNAME) \
+		TMPDIRNAME=$(TMPDIRNAME) \
+		CHPL_MAKE_RUNTIME_LIB=$(CHPL_MAKE_RUNTIME_LIB) \
+		CHPL_MAKE_RUNTIME_INCL=$(CHPL_MAKE_RUNTIME_INCL) \
+		CHPL_MAKE_THIRD_PARTY=$(CHPL_MAKE_THIRD_PARTY) \
+		CHPL_MAKE_IS_MLI=1
+#
+# Compile the client bundle. Exactly the same as for static compilation, the
+# only thing that changes here are the flags inserted by codegen_makefile.
+#
+	$(CC) $(CHPL_MAKE_BASE_CFLAGS) $(GEN_CFLAGS) $(COMP_GEN_CLIENT_CFLAGS) \
+		-c -o $(TMPBINNAME).o \
+		$(CHPLSRC) \
+		$(COMP_GEN_MLI_EXTRA_INCLUDES) \
+		$(CHPL_RT_INC_DIR) \
+		$(CHPL_MLI_INCS)
+#
+# Invoke the linker directly rather than package everything into an archive.
+# Again, the codegen_makefile function is responsible for inserting the
```

Comment:
Re: my feedback above, if `Makefile.mli-*` threw `$LIB_\(STATIC\|DYNAMIC\)_FLAG` themselves here, what was happening would be clearer and we wouldn't need this comment at all.